coral/src/app/features/configuration/clusters/AddNewClusterForm.test.tsx

```diff
@@ -0,0 +1,291 @@
+import { cleanup, screen, within } from "@testing-library/react";
+import { userEvent } from "@testing-library/user-event";
+import AddNewClusterForm from "src/app/features/configuration/clusters/AddNewClusterForm";
+import { customRender } from "src/services/test-utils/render-with-wrappers";
+import { addNewCluster } from "src/domain/cluster";
+
+jest.mock("src/domain/cluster/cluster-api.ts");
+
+const mockAddNewClusterRequest = addNewCluster as jest.MockedFunction<
+  typeof addNewCluster
+>;
+
+const mockedUsedNavigate = jest.fn();
+jest.mock("react-router-dom", () => ({
+  ...jest.requireActual("react-router-dom"),
+  useNavigate: () => mockedUsedNavigate,
+}));
+
+const mockedUseToast = jest.fn();
+jest.mock("@aivenio/aquarium", () => ({
+  ...jest.requireActual("@aivenio/aquarium"),
+  useToast: () => mockedUseToast,
+}));
+
+describe("<AddNewClusterForm />", () => {
+  const originalConsoleError = console.error;
+
+  beforeEach(() => {
+    console.error = jest.fn();
+
+    customRender(<AddNewClusterForm />, {
+      queryClient: true,
+      aquariumContext: true,
+    });
+  });
+  afterEach(() => {
+    console.error = originalConsoleError;
+
+    cleanup();
+    jest.clearAllMocks();
+  });
+
+  it("renders all necessary elements by default", async () => {
```

Comment:
Bit of a nitpick (and no need to change that here), but usually, I try to keep the pattern of having one assertion per `it` block, so it's fast and telling when one assertion fails. So more having "renders all necessary elements" as a `describe` block & every elements query & assertion in one `it` block. It also adds more information value to tests.